### ISSUE TITLE
[Bug]: native iterators don't survive the contextbridge, but Symbol.iterator does

### ISSUE THREAD

**Issue by MareikeTaeubner** [electron/electron#37504]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

23.1.0

### What operating system are you using?

Windows

### Operating System Version

Windows 10 Pro Version	10.0.19044 Build 19044

### What arch are you using?

x64

### Expected Behavior

We expected native iterables to survive the contextbridge, because custom iterable implementations do.

preload
```js
contextBridge.exposeInMainWorld("test_api", {
  iterate(iterable) {
    const values = [];
    for (const item of iterable) {
      values.push(item);
    }
    return values;
  }
});
```

renderer

```js
   function* generate() {
      for (const i of [1, 2, 3]) {
        yield i;
      }
    }

    const iterable = generate();

    const result = window.test_api.iterate(iterable);
```

## Full example
See the full working example including test cases [here](https://github.com/MareikeTaeubner/mcve-contextbridge-symbols)

## What we try to do
We try to invoke a function `iterate` defined in the preload script with a parameter that is an iterator from the renderer process:

```js
contextBridge.exposeInMainWorld("test_api", {
  iterate(iterable) {
    const values = [];
    for (const item of iterable) {
      values.push(item);
    }
    return values;
  }
});
```
## What we see
If we call this function with a native iterable from the renderer process, it will fail with `Error: iterable is not iterable`:

```js
   function* generate() {
      for (const i of [1, 2, 3]) {
        yield i;
      }
    }

    const iterable = generate();

    const result = window.test_api.iterate(iterable);
  ```
  
 However, if we implement the iterable ourselves, it will work. We see that, contrary to what the [documentation](https://www.electronjs.org/docs/latest/api/context-bridge#parameter--error--return-type-support) says, the `Symbol.iterator` survives.
 
 ```js
const values = [1, 2, 3];
    let index = 0;

    const iterable = {
      [Symbol.iterator]: function () {
        return {
          next() {
            if (index >= values.length) {
              return { done: true };
            } else {
              return { value: values[index++] };
            }
          },
        };
      },
    };
    
    const result = window.test_api.iterate(iterable);

```
    

### Testcase Gist URL

[test sample as electron fiddle gist](https://gist.github.com/7f45335a82bdd797c439b9067ca723b4)

**Comment by MareikeTaeubner** [comment]
I added the sample additionally as electron fiddle gist. If something is still missing, I would be glad if you tell me.

**Comment by codebytere** [comment]
cc @MarshallOfSound

**Comment by MareikeTaeubner** [comment]
Thank you all so much for fixing this 👍

**Comment by MareikeTaeubner** [comment]
@codebytere unfortunately I cannot verify the fix with the nightly build v25.0.0-nightly.20230329

**Comment by Arimanecro** [comment]
Version "25.1.0"
Same error:
`Uncaught Error: An object could not be cloned`
